File: UniData/U2/uonet/samples/C#/UniDynArray/UniDynArraySample.cs
using System;
using IBMU2.UODOTNET;

namespace IBMU2.Samples.UniDynArraySample
{
	/// <summary>
	/// Summary description for Class1.
	/// </summary>
	class UniDynArraySample
	{
		/// <summary>
		/// The main entry point for the application.
		/// </summary>
		[STAThread]
		static void Main(string[] args)
		{
			UniSession us1=null;
			
			try
			{
				us1 = UniObjects.OpenSession("localhost","ZZZ","xxxx","HS.SALES","uvcs");
				
				//creating UniDynArray
				char bFM = Convert.ToChar(254);
				char bVM = Convert.ToChar(253);
				char bSVM = Convert.ToChar(252);

				UniDynArray lDynArray =  new UniDynArray(us1,"ab" + bFM + "cd" + bVM + "ef" + bVM
										+ "gh" + bVM + "ij" + bFM + "kl" + bSVM + "mn" + bSVM + "no" + 
										bVM + "p" + bVM + "qr" + bFM + "s" + bFM + "t" + bFM + "");
				
				// run  Count()
				int myVal = lDynArray.Count();
				
				// run Dcount()
				int myVal2 = lDynArray.Dcount();

				// run Extract 
				UniDynArray real = lDynArray.Extract(1,1,0);

				// run Replace
				lDynArray.Replace(2, 0, 0, "*");

				//run delete
				lDynArray.Delete(1, 0, 0);

				// run insert
				lDynArray.Insert(0, 0, 0, "2500");
			}
			catch(Exception e)
			{
				if(us1 != null && us1.IsActive)
				{
					UniObjects.CloseSession(us1);
					us1= null;
				}
				Console.WriteLine("");
				string s = "Connection Failed : " + e.Message;
				Console.WriteLine(s);
			}
			finally
			{
				if(us1 != null && us1.IsActive)
				{
					Console.WriteLine("");
					string s = "Connection Passed";
					Console.WriteLine(s);
					UniObjects.CloseSession(us1);
				}
				
			}
		}
	}
}

File: UniData/U2/uonet/samples/VB.NET/UniDynArray/UniDynArraySample.vb
Option Explicit On 
Option Strict On

' Add the classes in the following namespaces to our namespace
Imports System
Imports System.IO
Imports System.Text
Imports System.Collections
Imports IBMU2.UODOTNET


Class App


    Public Shared Function Main(ByVal args() As String) As Integer

        Dim us1 As UniSession = Nothing
        Try
            Console.WriteLine("Program started... ")
            us1 = UniObjects.OpenSession("localhost", "ZZZ", "xxxx", "HS.SALES", "uvcs")

            '/creating UniDynArray
            Dim bFM As Char = Convert.ToChar(254)
            Dim bVM As Char = Convert.ToChar(253)
            Dim bSVM As Char = Convert.ToChar(252)
            Dim str As String
            str = "ab" & bFM & "cd" & bVM & "ef" & bVM & "gh" & bVM & "ij" & bFM & "kl" & bSVM & "mn" & bSVM & "no" & bVM & "p" & bVM & "qr" & bFM & "s" & bFM & "t" & bFM & ""
            Dim lDynArray As New UniDynArray(us1, str)

            ' run  Count()
            Dim myVal As Integer = lDynArray.Count()

            ' run Dcount()
            Dim myVal2 As Integer = lDynArray.Dcount()

            ' run Extract 
            Dim real As UniDynArray = lDynArray.Extract(1, 1, 0)

            'run Replace
            lDynArray.Replace(2, 0, 0, "*")

            'run delete
            lDynArray.Delete(1, 0, 0)

            ' run insert
            lDynArray.Insert(0, 0, 0, "2500")

        Catch e As Exception
            If Not (us1 Is Nothing) Then
                If (us1.IsActive) Then
                    UniObjects.CloseSession(us1)
                    us1 = Nothing
                End If
            End If
            Console.WriteLine("")
            Dim s As String
            s = "Connection Failed : " + e.Message
            Console.WriteLine(s)
        Finally
            If Not (us1 Is Nothing) Then
                If (us1.IsActive) Then
                    Console.WriteLine("")
                    Dim s As String
                    s = "Connection Passed"
                    Console.WriteLine(s)
                    UniObjects.CloseSession(us1)
                    Console.WriteLine("Program finished... ")
                End If

            End If

        End Try
        Console.Read()
        Return 0
    End Function 'Main
End Class 'App
